KFSPTS-25511 Add document for Jaggaer link mapping

cah292 opened the pull request:
There are PRs for this change in both cu-kfs and nonprod-sql. Please make sure both PRs are good to go before merging.

This PR adds a new maintenance document for adding/editing Jaggaer Role Link Mapping BOs, and also adds the related lookup and inquiry screens.

The lookup and inquiry metadata has generally been structured to use the newer frameworks, but for consistency with KualiCo's other converted BOs, the new inquiry has been configured to allow for both the old and new setup. We can make further revisions in the future once KualiCo fully retires the old framework, but if you think we should not use the old inquiry framework here at all, please let me know.

jhulslander commented on src/main/resources/edu/cornell/kfs/module/purap/businessobject/datadictionary/JaggaerRoleLinkMapping.xml:
```diff
@@ -0,0 +1,171 @@
+          abstract="true"
+          parent="GenericAttributes-activeIndicator"
+          p:name="eShopLink"
+          p:label="Assignable on eShop Login Link"
```
I'm not sure about "Assignable" here.  What about "Available" on eShop link?

cah292 replied:
I'll consult with the functionals on the naming like you suggested.

jhulslander commented on src/main/resources/edu/cornell/kfs/module/purap/businessobject/datadictionary/JaggaerRoleLinkMapping.xml:
```diff
@@ -0,0 +1,171 @@
+          abstract="true"
+          parent="GenericAttributes-activeIndicator"
+          p:name="contractsPlusLink"
+          p:label="Assignable on Contracts+ Login Link"
```
Similar thought there, perhaps "Available" on Contract Plus LInk?

cah292 replied:
Will ask functionals.

jhulslander commented on src/main/resources/edu/cornell/kfs/module/purap/businessobject/datadictionary/JaggaerRoleLinkMapping.xml:
```diff
@@ -0,0 +1,171 @@
+          abstract="true"
+          parent="GenericAttributes-activeIndicator"
+          p:name="jaggaerAdminLink"
+          p:label="Assignable on Jaggaer Admin Login Link"
```
similar thought here too

cah292 replied:
Will ask functionals.

jhulslander reviewed (changes requested):
This change look good for the most part.  I noted a few labels which I am not sure about. Perhaps I am over thinking it.  It might be best to get feedback from functionals to see what they think.

As for inclusion of both the old new inquiry frameworks.  is there a reason why we would need to use told framework?  To avoid redundancy and keep thing as simple as possible, I think we should just include the new framework, unless there is a technical reason we need to include the old one.

cah292 commented:
I had kept in the old inquiry setup for consistency with KualiCo' setup, but I agree that we don't need it if the new inquiry runs fine without it. I'll try removing the old inquiry setup and see if things still work.

cah292 commented:
I've refactored the DD XML to remove the unneeded legacy inquiry beans, and I also updated the checkbox labels based on Jay's suggestion after confirming with the functionals. This is ready for review again.

jhulslander commented on src/main/resources/edu/cornell/kfs/module/purap/document/workflow/JaggaerRoleLinkMappingMaintenanceDocument.xml:
```diff
@@ -0,0 +1,12 @@
+<data xmlns="ns:workflow" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="ns:workflow resource:WorkflowData">
+    <documentTypes xmlns="ns:workflow/DocumentType" xsi:schemaLocation="ns:workflow/DocumentType resource:DocumentType">
+        <documentType>
+            <name>JRLM</name>
```
sorry for not mentioning this before, but rather than ingesting XML, could this doc type be done by inserting a new record into kfs.krew_doc_typ_t?

cah292 replied:
Yes, based on my review of Nancy's changes on a different PR, I believe I could insert the doc type with SQL just by inserting a record into KREW_DOC_TYP_T, due to the type of setup being inherited by the new doc type. I will look into adding that SQL.

jhulslander replied:
Can this be removed if you are inserting the document type with SQL?

cah292 replied:
For our Cornell-specific doc types, I thought we still kept their workflow XML in our codebase for reference, even when using SQL to insert them manually.

jhulslander replied:
hmmm, I would think the reverse, lets check with the team

jhulslander replied:
based on discussion in teams, we do want to keep this

jhulslander reviewed (commented):
Changes look good, I just had one question if it would make more sense to add the doc type with SQL rather than workflow XML.

cah292 commented:
I've committed some SQL on the related PR to auto-insert the new doc type. This is ready for review again.

jhulslander reviewed (approved):
Unit tests pass, SQL is ready, I shall merge
